### PR TITLE
feat: Paginate User Table

### LINKED ISSUE #1262: 🔧 [Refactor] apollo anfrage  für unregistered e-mails
## 🔧 Refactor ticket
Wenn man auf den 'button'  Unregistered e-mails klicke, wird  momentan nur aus der Liste gefiltert welche schon von der datenbank abgerufen wurde und filtert die user die keine registrierte mail haben heruas. 

Aufgabe:
Bitte ändere das Event auf diesem button so das eine apollo anfarge gestellt wird und nur nutzer zurückgegeben werden welche ihre e-mail noch nicht regierstert haben. 

 Page: admin/src/pages/UserSearch.vue
 Methods: 
`unconfirmedRegisterMails() {
      this.searchResult = this.searchResult.filter((user) => {
        return user.emailChecked
      })
    },`
 
FireShot Capture 900 - Gradido Admin Interface - localhost

Es gibt ein PR mit dem filter: https://github.com/gradido/gradido/pull/1256

### LINKED ISSUE #1275: 🐛 [Bug] Unselect Unregistered Emails
## 🐛 Bugreport
When clicking on unregistered emails in the admin interface, there is no way to switch off the filter again.

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest
Add pagination buttons to the user table. Filter unchecked emails on server.

- fixes #1275
- fixes #1262

### CODE REVIEW

File: admin/src/pages/Creation.vue [hunk 1]
```diff
@@ -18,6 +18,13 @@
           :creation="creation"
           @update-item="updateItem"
         />
+        <b-pagination
+          pills
+          v-model="currentPage"
+          per-page="25"
+          :total-rows="rows"
+          align="center"
+        ></b-pagination>
```
Comment (Elweyn) [inline admin/src/pages/Creation.vue:27]: This is only for me as question to understand:
Why do you create a b-pagination? Is it a bootstrap component that shows the page on which we are and the next pages?

File: backend/src/middleware/klicktippMiddleware.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ import CONFIG from '../config/index'
 //   // Do Something here before resolver is called
 //   const result = await next()
 //   // Do Something here after resolver is completed
-//   await signIn(result.email, result.language, result.firstName, result.lastName)
+//   await klicktippSignIn(result.email, result.language, result.firstName, result.lastName)
```
Comment (Elweyn) [inline backend/src/middleware/klicktippMiddleware.ts:15]: Doesn't we need this anymore?
When did it get commented out?

File: database/src/index.ts [hunk 3]
```diff
@@ -45,6 +46,10 @@ const run = async (command: string) => {
       await runSeeder(CreateBibiBloxbergSeed)
       await runSeeder(CreateRaeuberHotzenplotzSeed)
       await runSeeder(CreateBobBaumeisterSeed)
+      // eslint-disable-next-line prefer-spread
+      Array.apply(null, Array(96)).forEach(async () => {
+        await runSeeder(CreateUserSeed)
+      })
```
Comment (Elweyn) [inline database/src/index.ts:52]: What is this for? Create 96 new Users?

**Review by Elweyn — APPROVED** [review]

File: admin/src/pages/Creation.vue (hunk 1 above)
Comment (Mogge) [reply admin/src/pages/Creation.vue:27]: https://bootstrap-vue.org/docs/components/pagination

File: backend/src/middleware/klicktippMiddleware.ts (hunk 2 above)
Comment (Mogge) [reply backend/src/middleware/klicktippMiddleware.ts:15]: I do not know. This is your code.

File: database/src/index.ts (hunk 3 above)
Comment (Mogge) [reply database/src/index.ts:52]: Yes

File: admin/src/locales/de.json [hunk 4]
```diff
@@ -1,4 +1,5 @@
 {
+  "all_emails": "Alle E-Mails",
```
Comment (ulfgebhardt) [inline admin/src/locales/de.json:2]: Its All users - the page is the user search - consider to change this

File: admin/src/locales/de.json [hunk 5]
```diff
@@ -53,7 +54,7 @@
   "transactionlist": {
     "title": "Alle geschöpften Transaktionen für den Nutzer"
   },
-  "unregistered_emails": "Unregistrierte E-Mails",
+  "unregistered_emails": "Nur unregistrierte E-Mails",
```
Comment (ulfgebhardt) [inline admin/src/locales/de.json:57]: Same here

File: admin/src/pages/Creation.vue [hunk 6]
```diff
@@ -18,6 +18,13 @@
           :creation="creation"
           @update-item="updateItem"
         />
+        <b-pagination
+          pills
+          v-model="currentPage"
+          per-page="25"
```
Comment (ulfgebhardt) [inline admin/src/pages/Creation.vue:24]: this should be a parameter

File: admin/src/graphql/searchUsers.js [hunk 7]
```diff
@@ -1,14 +1,17 @@
...
+    searchUsers(searchText: $searchText, currentPage: $currentPage, notActivated: $notActivated) {
+      userCount
+      userList {
+        userId
+        firstName
+        lastName
+        email
+        creation
+        emailChecked
+      }
```
Comment (ulfgebhardt) [inline admin/src/graphql/searchUsers.js:14]: the page size should be a parameter and a globally defined constant for now - else we have to change it in trillions of places

File: admin/src/pages/UserSearch.vue [hunk 8]
```diff
@@ -21,6 +21,14 @@
       :fieldsTable="fields"
       :criteria="criteria"
     />
+    <b-pagination
+      pills
+      size="lg"
+      v-model="currentPage"
+      per-page="25"
```
Comment (ulfgebhardt) [inline admin/src/pages/UserSearch.vue:28]: parameter pls

File: backend/jest.config.js [hunk 9]
```diff
@@ -1,15 +1,18 @@
...
-  collectCoverageFrom: ['src/**/*.ts', '!**/node_modules/**'],
-  moduleNameMapper: {
-    '@entity/(.*)': '<rootDir>/../database/build/entity/$1',
-    // This is hack to fix a problem with the library `ts-mysql-migrate` which does differentiate between its ts/js state
-    '@dbTools/(.*)':
-      process.env.NODE_ENV === 'development'
-        ? '<rootDir>/../database/src/$1'
-        : '<rootDir>/../database/build/src/$1',
-  },
+module.exports = async () => {
```
Comment (ulfgebhardt) [inline backend/jest.config.js:2]: why do you do this `async`? This looks very dangerous?!

File: backend/src/graphql/resolver/AdminResolver.ts [hunk 10]
```diff
@@ -26,11 +27,13 @@ import { LoginUserRepository } from '../../typeorm/repository/LoginUser'
...
-  @Query(() => [UserAdmin])
-  async searchUsers(@Arg('searchText') searchText: string): Promise<UserAdmin[]> {
+  @Query(() => SearchUsersResult)
+  async searchUsers(
+    @Args() { searchText, currentPage = 1, pageSize = 25, notActivated = false }: SearchUsersArgs,
+  ): Promise<SearchUsersResult> {
     const userRepository = getCustomRepository(UserRepository)
     const users = await userRepository.findBySearchCriteria(searchText)
-    const adminUsers = await Promise.all(
+    let adminUsers = await Promise.all(
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/AdminResolver.ts:36]: I see whats happening here - this is ugly - please add a todo - we MUST refactor this

File: database/src/index.ts (hunk 3 above)
Comment (ulfgebhardt) [reply database/src/index.ts:52]: Yes wtf?

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**